crates/constraint-vm/src/access.rs

```diff
@@ -286,42 +294,92 @@ pub(crate) fn repeat_counter(stack: &mut Stack, repeat: &Repeat) -> OpResult<()>
     Ok(stack.push(counter)?)
 }
 
-pub(crate) fn transient(stack: &mut Stack, solution: SolutionAccess) -> OpResult<()> {
-    let pathway = stack.pop()?;
-    let pathway =
-        SolutionDataIndex::try_from(pathway).map_err(|_| AccessError::TransientDataOutOfBounds)?;
-    let value = stack.pop_len_words::<_, _, StackError>(|key| {
-        let value = solution
-            .transient_data
-            .get(&pathway)
-            .ok_or(StackError::IndexOutOfBounds)?
+pub(crate) fn pub_var(stack: &mut Stack, pub_vars: &TransientData) -> OpResult<()> {
+    // Pop the value_len, value_ix and create a range.
+    let value_len = stack
+        .pop()
+        .map_err(|_| MissingAccessArgError::PubVarValueLen)
+        .map_err(AccessError::from)?;
+    let value_ix = stack
+        .pop()
+        .map_err(|_| MissingAccessArgError::PubVarValueIx)
+        .map_err(AccessError::from)?;
+    let range = range_from_start_len(value_ix, value_len).ok_or(AccessError::InvalidAccessRange)?;
+
+    // Get the pathway_ix.
+    // Note this does not pop the pathway_ix from the stack.
+    let pathway_ix = stack
+        .skip_len_words()
```

Comment:
Yeh that's the way it was but it felt wrong as it's basically keyed as `pathway_id, key => value` but maybe I'm looking at the stack backwards. As in top of the stack is actually the first parameter to a function.
But then it would make more sense to do:
```
[value_len, value_ix, key, key_len, pathway_ix]
```
which would be equivalent to a function call `pub_var(pathway_ix, key, value_ix, value_len)`.

It just looks weird in the asm docs because it's written in `[stack_bottom, stack_top]`.